src/main/java/io/gravitee/policy/messagefiltering/MessageFilteringPolicy.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.policy.messagefiltering;
+
+import io.gravitee.gateway.jupiter.api.context.MessageExecutionContext;
+import io.gravitee.gateway.jupiter.api.message.Message;
+import io.gravitee.gateway.jupiter.api.policy.Policy;
+import io.gravitee.policy.messagefiltering.configuration.MessageFilteringPolicyConfiguration;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Maybe;
+import lombok.AllArgsConstructor;
+
+/**
+ * @author David BRASSELY (david.brassely at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@AllArgsConstructor
+public class MessageFilteringPolicy implements Policy {
+
+    private final MessageFilteringPolicyConfiguration configuration;
+
+    @Override
+    public String id() {
+        return "message-filtering";
+    }
+
+    @Override
+    public Completable onMessageRequest(MessageExecutionContext ctx) {
+        return Completable.defer(
+            () -> {
+                String computedFilter = computeFilter(ctx);
```

Comment:
I would have put the call to computeFilter() inside filter() method as it's the first required step for the filter to compute what we have in the configuration correctly

but it's not a blocker

Reply:
I don"t understand you comment. computeFilter is not about filtering the ctx but is about computing the EL filter expression. Does it make sense?

Reply:
OK after review I got it. I don't want to compute the filter on each message, only once for all. That why the computation is outside the filter method.

Reply:
yeah indeed, I understand better.